File: src/refactoring-exercise/code-replacer/dot-net/RefactoringExample/RefactoringExample/CodeReplacer.cs
// Replace %CODE% with requested id, and %ALTCODE% with "dashed" version of id.
using System;
using System.IO;
using System.Reflection;
using System.Text;

namespace RefactoringExample
{
    public class CodeReplacer
    {
        public String SourceTemplate { get; set; }
        
        String _code;
        String _altcode;

        public void Substitute(String reqId, TextWriter output)
        {
            // Read in the template file
            StringBuilder sb = new StringBuilder("");

            try
            {
                Assembly _assembly = Assembly.GetExecutingAssembly();
                StreamReader sr = new StreamReader(_assembly.GetManifestResourceStream("RefactoringExample.template.html"));
                String line;
                while (((line = sr.ReadLine()) != "") && (line != null))
                    sb = new StringBuilder(sb + line + "\n");
                sr.Close();
            }
            catch (Exception e)
            {
            }
            SourceTemplate = sb.ToString();

            try
            {
                String template = (string)SourceTemplate.Clone();
                // Substitute for %CODE%
                int templateSplitBegin = template.IndexOf("%CODE%");
                int templateSplitEnd = templateSplitBegin + 6;
                String templatePartOne = template.Substring(0, templateSplitBegin);
                String templatePartTwo = template.Substring(templateSplitEnd, template.Length - 1);
                _code = reqId.ToString();
                template = templatePartOne + _code + templatePartTwo;

                // Substitute for %ALTCODE%
                templateSplitBegin = template.IndexOf("%ALTCODE%");
                templateSplitEnd = templateSplitBegin + 9;
                templatePartOne = template.Substring(0, templateSplitBegin);
                templatePartTwo = template.Substring(templateSplitEnd, template.Length - 1);
                _altcode = _code.Substring(0, 4) + "-" + _code.Substring(4, 4);
                output.Write(templatePartOne + _altcode + templatePartTwo);
            }
            catch (Exception e)
            {
                Console.WriteLine("Error in substitute()");
            }

            output.Flush();
            output.Close();
        }
    }
}

File: src/refactoring-exercise/code-replacer/java/src/main/CodeReplacer.java
import java.io.*;
import java.util.*;

// Replace %CODE% with requested id, and %ALTCODE% with "dashed" version of id.
// Run with -Dtemplatedir=src/resources/

public class CodeReplacer {
    public final String TEMPLATE_DIR = "templatedir";
    String sourceTemplate;
    String code;
    String altcode;

    public void substitute(String reqId, PrintWriter out) throws IOException 
    {
                // Read in the template file
        String templateDir = System.getProperty(TEMPLATE_DIR, "");
        StringBuffer sb = new StringBuffer("");
        try {
            FileReader fr = new FileReader(templateDir + "template.html");
            BufferedReader br = new BufferedReader(fr);
            String line;
            while(((line=br.readLine())!="")&&line!=null) sb = new StringBuffer(sb + line + "\n");
            br.close();
            fr.close();
        } catch (Exception e) {
        	e.printStackTrace();
        }
        sourceTemplate = new String(sb);

        try {
            String template = new String(sourceTemplate);    
                // Substitute for %CODE%
            int templateSplitBegin = template.indexOf("%CODE%");
            int templateSplitEnd = templateSplitBegin + 6;
            String templatePartOne = new String(template.substring(0, templateSplitBegin));
            String templatePartTwo = new String(template.substring(templateSplitEnd, template.length()));
            code = new String(reqId);
            template = new String(templatePartOne + code + templatePartTwo);

                // Substitute for %ALTCODE%
            templateSplitBegin = template.indexOf("%ALTCODE%");
            templateSplitEnd = templateSplitBegin + 9;
            templatePartOne = new String(template.substring(0, templateSplitBegin));
            templatePartTwo = new String(template.substring(templateSplitEnd, template.length()));
            altcode = code.substring(0,5) + "-" + code.substring(5,8);
            out.print(templatePartOne + altcode + templatePartTwo);
        } catch (Exception e) {
            System.out.println("Error in substitute()");
        }
        out.flush();
        out.close();
    }
}